lib/action_view/component/railtie.rb

```diff
@@ -25,14 +26,22 @@ class Railtie < Rails::Railtie # :nodoc:
         require "railties/lib/rails/components_controller"
         require "railties/lib/rails/component_examples_controller"
 
-        app.config.eager_load_namespaces << ActionView::Component::Base
+        #app.config.eager_load_namespaces << ActionView::Component::Base
         options = app.config.action_view_component
 
         if options.show_previews && options.preview_path
           ActiveSupport::Dependencies.autoload_paths << options.preview_path
         end
       end
 
+      initializer "action_view_component.eager_load_actions" do
+        ActiveSupport.on_load(:after_initialize) do
+          ActionView::Component::Base.descendants.each do |descendant|
+            descendant.compile if descendant.has_initializer? && config.eager_load
+          end
+        end
+      end
```

Comment:
this is a somewhat simpler way to handle compiling than polluting base with `eager_load!`. Not necessary for the fix but learned this along the way digging into other railties.

Reply:
@bostanio is this change functionally equivalent to what we had before?

Also, do we still need this line, then?

https://github.com/github/actionview-component/pull/158/files#diff-051fdf6a403587a2c819adf69b1ce4f5R28

Reply:
Yes I believe so and it passes the specs around eager load (and manually verified they fail if you turn eager_load off in test). If we upstream this into Rails they have some nice support for testing railties in dev/prod environments. I looked at recreating it but its a LOT of machinery.

No we don't need the line you references. I thought I'd removed it already but maybe missed a push.